File: basemodule/src/main/java/cn/wsgwz/basemodule/widgets/progress/ProgressLayout.java
package cn.wsgwz.basemodule.widgets.progress;

import android.view.View;

import java.util.List;

public interface ProgressLayout {

    public void showContent();

    public void showContent(List<Integer> skipIds);


    public void showLoading();

    public void showLoading(List<Integer> skipIds);


    public void showEmpty(View.OnClickListener buttonClickListener);

    public void showEmpty(View.OnClickListener buttonClickListener, List<Integer> skipIds);

    public void showEmpty(String description, View.OnClickListener buttonClickListener, List<Integer> skipIds);

    public void showEmpty(int icon, String description, View.OnClickListener buttonClickListener, List<Integer> skipIds);

    public void showEmpty(int icon, String description, String buttonText, View.OnClickListener buttonClickListener, List<Integer> skipIds);


    public void showError(View.OnClickListener buttonClickListener);
    public void showError(View.OnClickListener buttonClickListener, List<Integer> skipIds);

    public void showError(String description, View.OnClickListener buttonClickListener, List<Integer> skipIds);

    public void showError(int icon, String description, View.OnClickListener buttonClickListener, List<Integer> skipIds);

    public void showError(int icon, String description, String buttonText, View.OnClickListener buttonClickListener, List<Integer> skipIds);


    public String getCurrentState();

    public boolean isContentCurrentState();

    public boolean isLoadingCurrentState();

    public boolean isEmptyCurrentState();

    public boolean isErrorCurrentState();
}

File: basemodule/src/main/java/cn/wsgwz/basemodule/widgets/progressActivity/ProgressLayout.kt
package cn.wsgwz.basemodule.widgets.progressActivity

import android.view.View
import cn.wsgwz.basemodule.BaseApplication
import cn.wsgwz.basemodule.R

interface ProgressLayout {


    companion object {
        private const val TAG = "ProgressLayout"

        const val STATE_CONTENT = 1000
        const val STATE_LOADING = 1001
        const val STATE_EMPTY = 1002
        const val STATE_ERROR = 1003
    }





    fun showContent()

    fun showLoading()

    fun showEmpty(config: Config)
    fun showEmpty(l: (View) -> Unit) = showEmpty(Config.Builder().buttonClickListener(View.OnClickListener {
        l(it)
    }).description(BaseApplication.getInstance().getString(R.string.progress_layout_empty_description)).build())

    fun showError(config: Config)
    fun showError(l: (View) -> Unit) = showError(Config.Builder().buttonClickListener(View.OnClickListener {
        l(it)
    }).description(BaseApplication.getInstance().getString(R.string.progress_layout_error_description)).build())

    fun getCurrentState(): Int


    fun isContentCurrentState() = getCurrentState() == STATE_CONTENT
    fun isLoadingCurrentState() = getCurrentState() == STATE_LOADING
    fun isEmptyCurrentState() = getCurrentState() == STATE_EMPTY
    fun isErrorCurrentState() = getCurrentState() == STATE_ERROR




}